### PR TITLE
nrf: doc: Update TX injection documentation

### PR DESCRIPTION
**Pull request by VivekUppunda**
This documentation update handles the change in TX injection mode command becoming a network shell command from a Wi-Fi shell command

### CODE REVIEW

**Comment by sachinthegreen** [comment]
Adding @krga2022 so that STQA is aware of the feature updates.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Review by sachinthegreen — COMMENTED** [review]
I think an entry for `TX injection` would be needed in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/software_maturity.html#wi-fi-feature-support

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 1]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
 
 The nRF70 Series device supports raw 802.11 packet transmission when TX injection mode is enabled in the nRF Wi-Fi driver.
-To enable the TX injection mode of operation, you can use the ``NET_REQUEST_WIFI_MODE`` network management runtime API.
-This API can be used to disable TX injection mode in the driver when raw 802.11 packet transmission is not required by the application.
 
-TX injection mode can be enabled when the nRF70 Series device operates in Station mode, both when it is operating in a connected Station mode or in a non-connected Station mode (not connected to an Access Point (AP)).
+TX injection mode can be enabled when the nRF70 Series device primary mode of operation is set to Station mode or Monitor mode.
+When nRF70 Series Device is configured to Station Mode, TX injection mode can be enabled both when it is operating in a connected Station mode or in a non-connected Station mode (not connected to an Access Point (AP)).
+
+The ``net_eth_txinjection_mode`` functional API can be used to enable or disable TX injection mode in the nRF Wi-Fi driver as required by the application.
```
Comment (kittydepa) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:33]:
```suggestion
Use the``net_eth_txinjection_mode`` functional API to enable or disable TX injection mode.
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 2]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
+The ``NET_REQUEST_ETHERNET_SET_TXINJECTION_MODE`` network management API can also be used to enable or disable TX injection mode.
```
Comment (kittydepa) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:34]: Lines 34 - 35 can be combined to:
```suggestion
You can also use the ``NET_REQUEST_ETHERNET_SET_TXINJECTION_MODE`` network management API to enable or disable TX injection mode, however it is recommended to use the functional API (``net_eth_txinjection_mode``).
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 3]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
+However, it is recommended to use the functional API ``net_eth_txinjection_mode`` for enabling or disabling TX injection mode in the nRF Wi-Fi driver.
+
+The network management API ``NET_REQUEST_ETHERNET_GET_TXINJECTION_MODE`` can be used to obtain the current TX injection configured setting in the nRF Wi-Fi driver.
 
 When TX injection mode is enabled, you need to configure the operating channel.
-This channel will be used as the default transmit channel when the nRF70 device operates in a non-connected Station mode.
-When the device operates in Station mode connected to an AP on a given channel, the same channel will be used for raw 802.11 packet transmission.
+This channel will be used as the default transmit channel when the nRF70 Series device operates in a non-connected Station mode.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:40]: Suggestions here are fine, except keep the original ``operates``, as NCS docs lean towards an "active voice".

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 4]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
+When the nRF70 Series device operates in Station mode connected to an AP on a given channel, the same channel will be used for raw 802.11 packet transmission.
```
[the remark this answers is not in the document]
Comment (kittydepa) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:41]: Same comment as above, all suggestions fine but please keep ``operates``, unless this significantly changes the meaning and is required. Please ensure throughout. When possible "active voice" is preferred in NCS docs.

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 5]
```diff
@@ -26,22 +26,28 @@ TX injection mode
 *****************
 
 The nRF70 Series device supports raw 802.11 packet transmission when TX injection mode is enabled in the nRF Wi-Fi driver.
-To enable the TX injection mode of operation, you can use the ``NET_REQUEST_WIFI_MODE`` network management runtime API.
-This API can be used to disable TX injection mode in the driver when raw 802.11 packet transmission is not required by the application.
 
-TX injection mode can be enabled when the nRF70 Series device operates in Station mode, both when it is operating in a connected Station mode or in a non-connected Station mode (not connected to an Access Point (AP)).
+TX injection mode can be enabled when the nRF70 Series device primary mode of operation is set to Station mode or Monitor mode.
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: The UMAC team has not enabled this feature for SoftAP as far as i am aware of

[the remark this answers is not in the document]
Comment (sachinthegreen) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: @rajb9 / @karun2796 : Can you please look into this? As far as I understand `TX injection` is expected to work in all modes.

[the remark this answers is not in the document]
Comment (karun2796) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: The UMAC implementation does not restrict the Tx injection mode setting & sending Raw TX packet via any conditional checks (if_type). Thorough testing is still necessary to confirm correct operation.

[the remark this answers is not in the document]
Comment (VivekUppunda) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: So what does that mean @karun2796 ? Are you stating that UMAC has necessary checks for working in SoftAP mode ? As i understand with discussions with you, UMAC has various checks for different modes of operation and how TX injection works in those modes. SoftAP checks and interface code checks are implemented today?

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 6]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
+When nRF70 Series Device is configured to Station Mode, TX injection mode can be enabled both when it is operating in a connected Station mode or in a non-connected Station mode (not connected to an Access Point (AP)).
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:31]: Slightly modifying above suggestion:

```suggestion
When the nRF70 Series device is configured in Station mode, TX injection mode can be enabled both when the device is either connected to an Access Point (AP) or not connected to one.```

**Review by ioannisg — APPROVED** [review]
Looks good to me

File: doc/nrf/protocols/wifi/raw_tx_operation.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (ioannisg) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: Reading through this discussion, some points
- if Raw packet TX mode can safely work together with SoftAP, we probably need to demonstrate this in the same, along with adding this in the docs. This guide is primarily meant to show and explain what the stack has been validated to do, so it should be aligned with the sample where we show raw tx support :)
- Raw packet TX and SoftAP is probably not the most popular combination, so, if it helps, it is perfectly OK if the ambiguity around whether this combination of modes is resolved at post-merge (of this PR)

[the remark this answers is not in the document]
Comment (karun2796) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:30]: @VivekUppunda , yes previously we have some necessary checks and those checks got removed in last PR so i hope it should work fine with SoftAP too. But as i mentioned it need to be tested.

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 7]
```diff
@@ -26,22 +26,28 @@ TX injection mode
...
+However, it is recommended to use the functional API ``net_eth_txinjection_mode`` for enabling or disabling TX injection mode in the nRF Wi-Fi driver.
+
+The network management API ``NET_REQUEST_ETHERNET_GET_TXINJECTION_MODE`` can be used to obtain the current TX injection configured setting in the nRF Wi-Fi driver.
 
 When TX injection mode is enabled, you need to configure the operating channel.
-This channel will be used as the default transmit channel when the nRF70 device operates in a non-connected Station mode.
-When the device operates in Station mode connected to an AP on a given channel, the same channel will be used for raw 802.11 packet transmission.
+This channel will be used as the default transmit channel when the nRF70 Series device operates in a non-connected Station mode.
+When the nRF70 Series device operates in Station mode connected to an AP on a given channel, the same channel will be used for raw 802.11 packet transmission.
+When the nRF70 series device operates in Monitor mode, the configured channel for Monitor mode will be used as the transmit channel for TX injection operation.
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply doc/nrf/protocols/wifi/raw_tx_operation.rst:42]: Keeping the sentence as is as per previous comments for active case usage

**Review by sachinthegreen — APPROVED** [review]

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 8]
```diff
@@ -25,23 +25,28 @@ To use the raw packet transmit feature in your applications, you must enable the
...
 *****************
 
-The nRF70 Series device supports raw 802.11 packet transmission when TX injection mode is enabled in the nRF Wi-Fi® driver.
-To enable the TX injection mode of operation, you can use the ``NET_REQUEST_WIFI_MODE`` network management runtime API.
-This API can be used to disable TX injection mode in the driver when raw 802.11 packet transmission is not required by the application.
+The nRF70 Series device supports raw 802.11 packet transmission when TX injection mode is enabled in the nRF Wi-Fi driver.
 
-TX injection mode can be enabled when the nRF70 Series device operates in Station mode, both when it is operating in a connected Station mode or in a non-connected Station mode (not connected to an Access Point (AP)).
+TX injection mode can be enabled when the nRF70 Series device primary mode of operation is set to Station mode or Monitor mode.
+When the nRF70 Series device is configured in Station mode, TX injection mode can be enabled both when the device is either connected to an Access Point (AP) or not connected to one.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:31]: Please check that this suggestion still maintained the information you intended.
```suggestion
When the nRF70 Series device is configured in Station mode, TX injection mode can be enabled regardless of whether the device is connected to an Access Point (AP) or not.
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 9]
```diff
@@ -25,23 +25,28 @@ To use the raw packet transmit feature in your applications, you must enable the
...
+Use the ``net_eth_txinjection_mode`` functional API to enable or disable TX injection mode in the nRF Wi-Fi driver as required by the application.
+You can also use the ``NET_REQUEST_ETHERNET_SET_TXINJECTION_MODE`` network management API to enable or disable TX injection mode, however it is recommended to use the functional API (``net_eth_txinjection_mode``).
+
+The network management API ``NET_REQUEST_ETHERNET_GET_TXINJECTION_MODE`` can be used to obtain the current TX injection configured setting in the nRF Wi-Fi driver.
 
 When TX injection mode is enabled, you need to configure the operating channel.
-This channel will be used as the default transmit channel when the nRF70 device operates in a non-connected Station mode.
-When the device operates in Station mode connected to an AP on a given channel, the same channel will be used for raw 802.11 packet transmission.
+This channel will be used as the transmit channel when the nRF70 Series device operates in Station mode, but not connected to an AP.
+When the nRF70 Series device operates in Station mode and connected to an AP on a given channel, the connected channel will be used for raw 802.11 packet transmission.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:40]:
```suggestion
This channel will be used as the transmit channel when the nRF70 Series device operates in Station mode, but is not connected to an AP.
When the nRF70 Series device operates in Station mode and is connected to an AP on a given channel, the connected channel will be used for raw 802.11 packet transmission.
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 10]
```diff
@@ -25,23 +25,28 @@ To use the raw packet transmit feature in your applications, you must enable the
...
+When the nRF70 series device operates in Monitor mode, the configured channel for Monitor mode will be used as the transmit channel for TX injection operation.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:41]:
```suggestion
When the nRF70 Series device operates in Monitor mode, the configured channel for Monitor mode will be used as the transmit channel for TX injection operation.
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 11]
```diff
@@ -50,12 +55,34 @@ The following table lists the shell commands and network management APIs used to
...
-     - ``wifi mode -i1 -st``
-     - Configure interface instance 1 to Station with TX injection mode
+     - ``wifi mode -i1 -m``
+     - Configure interface instance 1 to Monitor mode
    * - net_mgmt(NET_REQUEST_WIFI_MODE)
      - ``wifi mode -i<interface instance> <configuration>``
      - ``wifi mode -i1 -s``
      - Configure interface instance 1 to Station mode
+
+The following table provides the parameter details for ``net_eth_txinjection_mode`` API which is used to enable/disable TX injection mode for raw packet transmission.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:65]:
```suggestion
The following table provides the parameter details for the ``net_eth_txinjection_mode`` API, which is used to enable or disable TX injection mode for raw packet transmission.
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 12]
```diff
@@ -50,12 +55,34 @@ The following table lists the shell commands and network management APIs used to
...
+
+.. list-table:: Wi-Fi raw packet transmission API parameter details
+   :header-rows: 1
+
+   * - API parameter
+     - description
+   * - iface
+     - Network interface on which TX injection mode is to be enabled
+   * - enable
+     - Boolean value to enable or disable TX injection mode.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:75]:
```suggestion
     - Boolean value to enable or disable TX injection mode
```

File: doc/nrf/protocols/wifi/raw_tx_operation.rst [hunk 13]
```diff
@@ -50,12 +55,34 @@ The following table lists the shell commands and network management APIs used to
...
+
+The following table provides network management API to set channel of operation for raw packet transmission.
```
Comment (richabp) [inline doc/nrf/protocols/wifi/raw_tx_operation.rst:77]:
```suggestion
The following table provides the network management API used to set the channel of operation for raw packet transmission.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Comment by VivekUppunda** [comment]
> I think an entry for `TX injection` would be needed in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/software_maturity.html#wi-fi-feature-support

As per discussion - adding BOARD_NRF7002DK_NRF5340_CPUAPP to software_maturity_features.yaml.
Thanks 
Vivek

**Review by kittydepa — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**